File: sol/solution/0001-0100/0008. String to Integer (atoi)/Solution.cs
public class Solution
{
    public int MyAtoi(string s)
    {
        int i = 0, n = s.Length, sign = 1, result = 0;
        while (i < n && s[i] == ' ') i++;
        if (i < n && (s[i] == '+' || s[i] == '-')) sign = (s[i++] == '-') ? -1 : 1;
        while (i < n && char.IsDigit(s[i]))
        {
            int digit = s[i++] - '0';
            if (result > (Int32.MaxValue - digit) / 10)
                return sign == 1 ? Int32.MaxValue : Int32.MinValue;
            result = result * 10 + digit;
        }
        return result * sign;
    }
}

File: sol/solution/0001-0100/0008. String to Integer (atoi)/Solution.scala
object Solution {
    def myAtoi(s: String): Int = {
        var i = 0
        var sign = 1
        var result = 0
        val n = s.length
        while (i < n && s(i) == ' ') i += 1
        if (i < n && (s(i) == '+' || s(i) == '-')) {
            sign = if (s(i) == '-') -1 else 1
            i += 1
        }
        while (i < n && s(i).isDigit) {
            val digit = s(i) - '0'
            if (result > (Int.MaxValue - digit) / 10)
                return if (sign == 1) Int.MaxValue else Int.MinValue
            result = result * 10 + digit
            i += 1
        }
        result * sign
    }
}